frontend/src/scenes/insights/utils.tsx

```diff
@@ -570,28 +570,36 @@ export function isQueryTooLarge(query: Node<Record<string, any>>): boolean {
     return queryLength > 1024 * 1024
 }
 
-export function parseDraftQueryFromLocalStorage(
-    query: string
-): { query: Node<Record<string, any>>; timestamp: number } | null {
+function parseAndMigrateQuery<T>(query: string): T | null {
```

Comment:
This should be added to the notebooks migration https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/notebooks/Notebook/migrations/migrate.ts#L44-L56 as well. Ideally there should also be a unit test.